app/analysis/edit/controller.js

```diff
@@ -226,4 +236,54 @@ export default class AnalysisEditController extends Controller.extend(
       changeset.rollback();
     });
   }
+
+  get queryParamsState() {
+    const states = {};
+    for (const param of this.queryParams) {
+      const defaultValue = getDefaultQueryParamValue(param);
+      const currentValue = this[param];
+      states[param] = {
+        as: param,
+        defaultValue,
+        serializedValue: currentValue,
+        value: currentValue,
+        changed: currentValue !== defaultValue,
+      };
+      if (["fromDate", "toDate"].includes(param)) {
+        states[param].serialize = serializeMoment;
+        states[param].deserialize = deserializeMoment;
+      }
+
+      if (param === "id") {
+        states[param].serialize = (arr) => {
+          return (arr && Array.isArray(arr) && arr.join(",")) || null;
+        };
+        states[param].deserialize = (str) => {
+          return (str && str.split(",")) || [];
+        };
+      }
+    }
+    return states;
+  }
+
+  resetQueryParams(...args) {
+    if (!args[0]) {
+      return;
+    }
+    const params = [...args[0]];
+    for (const param of params) {
+      this[param] = getDefaultQueryParamValue(param);
```

Comment:
Streamline with the other `resetQueryParams` methods. This is something you could extract into a general controller `QueryParamsController` since it's all the same.